mps-model-adapters/src/main/kotlin/org/modelix/model/mpsadapters/MPSLanguageRepository.kt

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2023.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.modelix.model.mpsadapters
+
+import jetbrains.mps.smodel.adapter.ids.SConceptId
+import jetbrains.mps.smodel.adapter.structure.MetaAdapterFactory
+import jetbrains.mps.smodel.language.ConceptRegistry
+import jetbrains.mps.smodel.language.LanguageRegistry
+import jetbrains.mps.smodel.runtime.illegal.IllegalConceptDescriptor
+import org.jetbrains.mps.openapi.module.SRepository
+import org.modelix.model.api.IConcept
+import org.modelix.model.api.ILanguageRepository
+
+class MPSLanguageRepository(private val repository: SRepository) : ILanguageRepository {
+    override fun resolveConcept(uid: String): IConcept? {
+        if (uid.startsWith("mps:")) return null
+
+        val conceptId = try {
+            SConceptId.deserialize(uid.substring(4))
+        } catch (e: Exception) { return null } ?: return null
```

Comment:
Maybe at least some logging in case of such a generic catch and also a comment why that's needed?

Reply:
How was this resolved?